src/main/java/org/spongepowered/api/text/channel/MessageReceiver.java

```diff
@@ -22,63 +22,51 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.text.sink;
+package org.spongepowered.api.text.channel;
 
-import org.spongepowered.api.command.CommandSource;
-
-import java.util.Set;
+import org.spongepowered.api.text.Text;
 
 /**
- * Factory for {@link MessageSink} instances.
+ * Represents something that can receive (and send) messages.
  */
-public interface MessageSinkFactory {
+public interface MessageReceiver {
 
     /**
-     * A message sink that targets all subjects with the given permission.
+     * Sends the formatted text message to source when possible. If text formatting
+     * is not supported in the implementation it will be displayed as plain text.
      *
-     * @param permission The permission to target
-     * @return The sink
+     * @param message The message
      */
-    MessageSink toPermission(String permission);
+    void sendMessage(Text message);
 
     /**
-     * A message sink that targets all subjects currently active.
+     * Sends the formatted text message(s) to source when possible. If text formatting
+     * is not supported in the implementation it will be displayed as plain text.
      *
-     * @return The sink
+     * @param messages The message(s)
      */
-    MessageSink toAll();
+    void sendMessages(Text... messages);
 
     /**
-     * A message sink that targets all players currently active.
+     * Sends the formatted text message(s) to source when possible. If text formatting
+     * is not supported in the implementation it will be displayed as plain text.
      *
-     * @return The sink
+     * @param messages The messages
      */
-    MessageSink toAllPlayers();
+    void sendMessages(Iterable<Text> messages);
 
     /**
-     * An empty message sink.
+     * Return the message channel that messages from this source should be sent to.
      *
-     * @return The sink
+     * @return This source's active message channel
      */
-    MessageSink toNone();
+    MessageChannel getMessageChannel();
```

Comment:
Wasn't there a discussion to move this in a separate interface `MessageSender` or similar?